### ISSUE TITLE
ServiceAccountCredentials.createScoped should return ServiceAccountCredentials

### ISSUE THREAD

**Issue by osi** [googleapis/google-auth-library-java#555]
https://github.com/googleapis/google-auth-library-java/blob/master/oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java#L516

method should return ServiceAccountCredentials so callers do not need to cast.

**Comment by elharo** [comment]
CLIRR doesn't like the change:

Error:  7006: com.google.auth.oauth2.ServiceAccountCredentials: Return type of method 'public com.google.auth.oauth2.GoogleCredentials createScoped(java.util.Collection)' has been changed to com.google.auth.oauth2.ServiceAccountCredentials

I wonder if that's a false positive?

**Comment by osi** [comment]
I appreciate the quick attention!

Interesting about the breakage. Searching doesn't yield any immediate answers. [This gist](https://gist.github.com/JakeWharton/14547311b1f145cb0695) seems to indicate that the compiler adds another bridge method, which would preserve compatibility? However, this [Stack Overflow](https://stackoverflow.com/a/3751291) answer indicates it may break subclasses.

**Comment by elharo** [comment]
Doesn't look like this can be done compatibly.

**Comment by osi** [comment]
understandable that your target is binary compatibility. thanks for trying!

**Comment by chingor13** [comment]
The general idea for this library is that you should only need to be working with any `GoogleCredentials` instance and the exact implementation you get is an implementation detail. In rare cases (explicit byte signing, Id token fetching), you may care if your `GoogleCredentials` instance implements a certain interface and thus you may need to cast.

**Comment by osi** [comment]
I encountered this when addressing deprecation warnings in `com.google.api.client.googleapis.auth.oauth2.GoogleCredential`

In our application, we prompt users to share Google Doc resources with the service account if the account does not have access. We were able to use `getServiceAccountId` to reflect the service account ID back to the user, so they could perform the share operation. 

Now, we need to know that we have a `ServiceAccountCredentials` instance to access `getClientEmail`.

When loading credentials, we do `ServiceAccountCredentials.fromStream().createScoped()`, and ran into this issue. 

I had started with `GoogleCredentials.fromStream()`, but moved to the sub-class directly when I realized I needed to care about the concrete type with the new API. There's no real problem with casting the result of `createScoped`; I was merely trying to avoid it, as it seemed avoidable. 

(All that to say, thanks for providing this. The story is merely context so you understand how/why it's being used 😀 )
